commands/utils/git.go

```diff
@@ -152,6 +151,14 @@ func (gm *GitManager) CreateBranchAndCheckout(branchName string) error {
 }
 
 func (gm *GitManager) createBranchAndCheckout(branchName string, create bool) error {
+	currentBranch, err := getCurrentBranch(gm.repository)
```

Comment:
i think maybe you can skip this check an just let Git to run the checkout and I believe that they do this check